src/main/java/net/minecraftforge/server/command/CommandTreeBase.java

```diff
@@ -42,16 +42,24 @@
  */
 public abstract class CommandTreeBase extends CommandBase
 {
-    private final HashMap<String, ICommand> commandMap = new HashMap<String, ICommand>();
+    private final Map<String, ICommand> commandMap = new HashMap<String, ICommand>();
 
     public void addSubcommand(ICommand c)
     {
         commandMap.put(c.getCommandName(), c);
     }
 
-    public Collection<ICommand> getSubCommands()
+    public Map<String, ICommand> getCommandMap()
     {
-        return Collections.unmodifiableCollection(commandMap.values());
+        return Collections.unmodifiableMap(commandMap);
+    }
+
+    public List<ICommand> getSortedCommandList()
+    {
+        List<ICommand> list = new ArrayList<ICommand>(commandMap.size());
```

Comment:
can just do `new ArrayList<ICommand>(commandMap.values());` instead of separate `addAll`